File: src/main/scala/no/bekk/scala/excercises/Lambda.scala
package no.bekk.scala.excercises

import java.time.LocalDate
import no.bekk.scala.model.{Player, League}

object Lambda {

  // TODO: Implement test
  def playersOlderThan(minAge: LocalDate, leagues: List[League]): List[Player] = {
    leagues.map(_.teams).reduce(_ ::: _).map(_.players).reduce(_ ::: _).filter(_.birthDate isAfter minAge)
  }
}

File: src/main/java/no/bekk/java/excercises/Lambda.java
package no.bekk.java.excercises;

import no.bekk.java.model.Player;
import no.bekk.java.model.Team;

import java.util.function.BinaryOperator;

public class Lambda {

	static BinaryOperator<Player> youngestPlayerLambda = (x, y) -> {
		if (x.birthday.isAfter(y.birthday)) return x;
		else return y;
	};

	static BinaryOperator<Team> highestTeamValueLambda = (x, y) -> {
		if (x.value > y.value) return x;
		else return y;
	};
}
